Mathlib/RingTheory/TwoSidedIdeal/Operations.lean

```diff
@@ -248,6 +244,31 @@ lemma mem_asIdealOpposite {I : TwoSidedIdeal R} {x : Rᵐᵒᵖ} :
   simpa [asIdealOpposite, asIdeal, TwoSidedIdeal.mem_iff, RingCon.op_iff] using
     ⟨I.ringCon.symm, I.ringCon.symm⟩
 
+/-- Bundle an `Ideal` that is already two-sided as a `TwoSidedIdeal`. -/
+def ofIdeal (I : Ideal R) (mul_mem_right : ∀ {x y}, x ∈ I → x * y ∈ I) : TwoSidedIdeal R :=
+  TwoSidedIdeal.mk' I I.zero_mem I.add_mem I.neg_mem (I.smul_mem _) mul_mem_right
+
+lemma mem_ofIdeal {I : Ideal R} {h} {x : R} :
+    x ∈ ofIdeal I h ↔ x ∈ I := by
+  simp [ofIdeal, mem_mk']
+
+@[simp]
+lemma coe_ofIdeal (I : Ideal R) (h) : (ofIdeal I h : Set R) = I := by
+  simp [ofIdeal]
+
+@[simp]
+lemma ofIdeal_asIdeal (I : TwoSidedIdeal R) (h) : ofIdeal (asIdeal I) h = I := by
+  ext
+  simp [mem_ofIdeal, mem_asIdeal]
+
+@[simp]
+lemma asIdeal_ofIdeal (I : Ideal R) (h) : asIdeal (ofIdeal I h) = I := by
```

Comment:
These could potentially be packaged as `TwoSidedIdeal R ≃o { I : Ideal R // mul_mem_right }`. However, I am not sure when (if ever) one would want to use that subtype instead of just `TwoSidedIdeal`.